Some unit tests fail with numpy 2.0.0

**Describe the bug**

As of 2024-06-21, a newly configured xcube environment gets numpy 2.0.0, released a few days ago with breaking API changes compared to the previous 1.26.4. Unsurprisingly, this breaks some xcube unit tests. Surprisingly, it only breaks five of them, so hopefully the required updates to xcube will be relatively minor. Summary of failures from [[https://ci.appveyor.com/project/bcdev/xcube/build/job/lw8fgi4bwcmcxi03?fullLog=true]]:

```text
FAILED test/core/store/test_descriptor.py::VariableDescriptorTest::test_variable_descriptor_to_dict - AttributeError: `np.NaN` was removed in the NumPy 2.0 release. Use `np.nan` instead.. Did you mean: 'nan'?
FAILED test/core/test_maskset.py::MaskSetTest::test_mask_set_with_flag_mask_int_array - AssertionError: 'c2rcc_flags(F1=(1, None), F2=(2, None), F3=(4, None), F4=(8, None))' != 'c2rcc_flags(F1=(np.int32(1), None), F2=(np.int32(2), None[46 chars]ne))'
- c2rcc_flags(F1=(1, None), F2=(2, None), F3=(4, None), F4=(8, None))
+ c2rcc_flags(F1=(np.int32(1), None), F2=(np.int32(2), None), F3=(np.int32(4), None), F4=(np.int32(8), None))
?                 +++++++++ +             +++++++ +++             +++++++++ +             +++++++++ +
FAILED test/core/test_maskset.py::MaskSetTest::test_mask_set_with_flag_mask_str - AssertionError: 'mask(sea=(1, None), land=(2, None), lake=(4, None), ice=(8, None))' != 'mask(sea=(np.uint8(1), None), land=(np.uint8(2), No[51 chars]ne))'
- mask(sea=(1, None), land=(2, None), lake=(4, None), ice=(8, None))
+ mask(sea=(np.uint8(1), None), land=(np.uint8(2), None), lake=(np.uint8(4), None), ice=(np.uint8(8), None))
?           +++++++++ +               +++++++++ +               +++++++++ +              +++++++ +++
FAILED test/core/test_maskset.py::MaskSetTest::test_mask_set_with_flag_values - AssertionError: 'SLC(no_data=(None, 0), saturated_or_defective=(None, 1), dar[271 chars]11))' != 'SLC(no_data=(None, np.uint8(0)), saturated_or_defective=(Non[391 chars]1)))'
Diff is 797 characters long. Set self.maxDiff to None to see it.
FAILED test/core/test_reproject.py::ReprojectTest::test_reproject_xy_to_wgs84_highroc - ValueError: dimensions () must have the same length as the number of data dimensions, ndim=1
```

**To Reproduce**
Steps to reproduce the behavior:
1. Create a fresh xcube environment (which will include numpy 2.0.0)
2. Run the test suite.
3. Observe the test failures described above.

**Expected behavior**
All unit tests pass.

**Screenshots**
n/a

**Additional context**
As ever, pinning numpy would provide a temporary solution, but it would (as ever in these cases) just create more work in the long run.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -91,6 +91,8 @@
 * Web API endpoints with an optional trailing slash are no longer listed
   twice in the automatically generated OpenAPI documentation (#965)
 
+* Several minor updates to make xcube compatible with NumPy 2.0.0 (#1024)
+
 ### Incompatible API changes
 
 * The `get_cmap()` method of `util.cmaps.ColormapProvider` now returns a 

--- a/test/core/store/test_descriptor.py
+++ b/test/core/store/test_descriptor.py
@@ -373,7 +373,7 @@ class VariableDescriptorTest(unittest.TestCase):
             "rj",
             ["dfjhrt", "sg"],
             chunks=(3, 2),
-            attrs={"ssd": 4, "zjgrhgu": "dgfrf", "fill_value": np.NaN},
+            attrs={"ssd": 4, "zjgrhgu": "dgfrf", "fill_value": np.nan},
         )
         expected = {
             "name": "xf",

--- a/xcube/core/maskset.py
+++ b/xcube/core/maskset.py
@@ -133,7 +133,9 @@ class MaskSet:
     def __str__(self):
         return "{}({})".format(
             self._flag_var.name,
-            ", ".join([f"{n}={v}" for n, v in self._flags.items()]),
+            ", ".join([f"{n}=({v[0]}, {v[1]})" for n, v in self._flags.items()]),
+            # We explicitly unpack the tuple to get the __str__ of the elements,
+            # rather than the __repr__. See https://peps.python.org/pep-3140/
         )
 
     def __dir__(self) -> Iterable[str]:

--- a/xcube/core/reproject.py
+++ b/xcube/core/reproject.py
@@ -394,8 +394,8 @@ def _ensure_valid_region(
         x1, y1, x2, y2 = region
     else:
         # Determine region from full-res lon/lat 2D variables
-        x1, x2 = x_var.min(), x_var.max()
-        y1, y2 = y_var.min(), y_var.max()
+        x1, x2 = x_var.min().item(), x_var.max().item()
+        y1, y2 = y_var.min().item(), y_var.max().item()
         extra_space = extra * max(x2 - x1, y2 - y1)
         # Add extra space in units of the source coordinates
         x1, x2 = x1 - extra_space, x2 + extra_space

